### PR TITLE
⚡️ Automated Leak Detection on CI ⚡️ 

### PR DESCRIPTION
**Pull request by samccone**
With this PR [Drool](https://github.com/samccone/drool)  is now providing us with the ability to fail CI when an implementation has memory leaks.

The perf tests are now run on every PR :space_invader: :spaghetti: 

1. Add automated leak detection :fountain: 
2. Only run deploy code paths on deploy :) (faster test runs for everyone) :horse: 
3. Add some more compiled src ignores

### CODE REVIEW

**Comment by samccone** [comment]
:green_apple:  CI is working (angular2 fails because of no routing)

https://travis-ci.org/tastejs/todomvc/builds/84436464

**Comment by samccone** [comment]
xref #1368

**Comment by ColinEberhardt** [comment]
Wow, that is awesome! I'll definitely recommend using Drool to the developers where I work.

**Comment by passy** [comment]
I love this! Just tested this with a few apps and it's sooo good! Thanks for this, @samccone!

File: tests/framework-path-lookup.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	list = list.filter(function (framework) {
+		return excludedFrameworks.indexOf(framework.name) === -1;
+	});
+
+	return list.filter(function (framework) {
+		return [].concat(names).some(function (f) {
+			return f === framework.name;
+		});
+	});
+}
```
Comment (arthurvr) [inline tests/framework-path-lookup.js]: Missing semicolon, why didn't the linting process catch this?

Comment (passy) [inline tests/framework-path-lookup.js]: Or me! Outrageous! Fixed in master with 1567e60.

Comment (arthurvr) [inline tests/framework-path-lookup.js]: Thanks @passy! Still we should check what's wrong with the linting tooling though.

**Pull request merged**